Assets/Scripts/HeadBob.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class HeadBob : MonoBehaviour
+{
+    public FpsController fpsController;
+
+    public float bobbingVerticalIntensity = 0.025f;
+    public float bobbingVerticalFrequency = 16f;
+
+    public float bobbingHorizontalIntensity = 0.015f;
+    public float bobbingHorizontalFrequency = 8f;
+
+    public float bobbingSpeed = 9f;
+
+    public float jumpBobIntensity = 0.25f;
+
+    private Vector3 _defaultLocalPosition;
+    private float _timer;
+
+
+    private void Awake()
+    {
+        _defaultLocalPosition = transform.localPosition;
+    }
+
+    private void Update()
+    {
+        var velocity = fpsController.velocity;
```

Comment:
You're only referencing the FpsController for the velocity. Suggest referencing the Rigidbody instead and removing the `velocity` field from the FpsController